quint/src/util.ts

```diff
@@ -68,3 +68,43 @@ export function findMap<X, Y>(xs: Iterable<X>, f: (x: X) => Maybe<Y>): Maybe<Y>
   }
   return none<Y>()
 }
+
+// Insert an item into an array sorted in ascending order by the given comparator.
+export function insertSorted<A>(array: A[], item: A, cmp: (a: A, b: A) => number): void {
+  if (array.length === 0) {
+    array.push(item)
+    return
+  }
+
+  const index = findInsertionIndex(array, x => cmp(x, item) > 0)
+  array.splice(index, 0, item)
+}
+
+/** Find the `index` in `array` where the predicate `pred(array[index])` is true,
+ * but `pred(array[index+1])` is false.
```

Comment:
This comment comes from a former version of this function and should be updated:

```suggestion
/** Find the `index` in `array` where `pred(array[index])` is true.
```